components/splitter/splitter.component.ts

```diff
@@ -0,0 +1,382 @@
+import { NzSplitterLayout } from './typings';
+import { coerceCollapsible, getPercentValue, isPercent } from './utils';
+
+interface PanelSize {
+  innerSize: number;
+  size: string | number | undefined;
+  hasSize: boolean;
+  postPxSize: number;
+  percentage: number;
+  min: string | number | undefined;
+  max: string | number | undefined;
+  postPercentMinSize: number;
+  postPercentMaxSize: number;
+}
+
+@Component({
+  selector: 'nz-splitter',
+  exportAs: 'nzSplitter',
+  preserveWhitespaces: false,
+  encapsulation: ViewEncapsulation.None,
+  template: `
+    <ng-template>
+      <ng-content></ng-content>
+    </ng-template>
+
+    @for (panel of panelProps(); let i = $index; track i; let last = $last) {
+      @let size = sizes()[i];
+      @let flexBasis = !!size.size ? size.size : 'auto';
+      @let flexGrow = !!size.size ? 0 : 1;
+      <div
+        class="ant-splitter-panel"
+        [class.ant-splitter-panel-hidden]="size.postPxSize === 0"
+        [style.flex-basis]="flexBasis"
+        [style.flex-grow]="flexGrow"
+      >
+        <ng-container *ngTemplateOutlet="panel.contentTemplate"></ng-container>
+      </div>
+
+      @if (!last) {
+        <div
+          nz-splitter-bar
+          [ariaNow]="size.percentage * 100"
+          [ariaMin]="size.postPercentMinSize * 100"
+          [ariaMax]="size.postPercentMaxSize * 100"
+          [resizable]="panel.resizable"
+          [collapsible]="panel.collapsible"
+          [active]="movingIndex() === i"
+          [vertical]="nzLayout() === 'vertical'"
+          [lazy]="nzLazy()"
+          [constrainedOffset]="constrainedOffset()"
+          (offsetStart)="startResize(i, $event)"
+          (collapse)="collapse(i, $event)"
+        ></div>
+      }
+    }
+
+    <!-- Fake mask for cursor -->
+    @if (movingIndex() !== null) {
+      <div
+        aria-hidden
+        class="ant-splitter-mask"
+        [class.ant-splitter-mask-horizontal]="nzLayout() === 'horizontal'"
+        [class.ant-splitter-mask-vertical]="nzLayout() !== 'horizontal'"
+      ></div>
+    }
+  `,
+  imports: [NgTemplateOutlet, NzSplitterBarComponent],
+  providers: [NzDestroyService],
+  host: {
+    class: 'ant-splitter',
+    '[class.ant-splitter-horizontal]': 'nzLayout() === "horizontal"',
+    '[class.ant-splitter-vertical]': 'nzLayout() !== "horizontal"'
+  }
+})
+export class NzSplitterComponent {
+  /** ------------------- Props ------------------- */
+  readonly nzLayout = input<NzSplitterLayout>('horizontal');
+  readonly nzLazy = input(false, { transform: booleanAttribute });
+  readonly nzResizeStart = output<number[]>();
+  readonly nzResize = output<number[]>();
+  readonly nzResizeEnd = output<number[]>();
+
+  readonly destroy$ = inject(NzDestroyService);
+  readonly elementRef = inject<ElementRef<HTMLElement>>(ElementRef);
+
+  /** ------------------- Panels ------------------- */
+  // Get all panels from content children
+  readonly panels = contentChildren(NzSplitterPanelComponent);
```

Comment:
`panels` can be private since it's used inside of the component itself only.  
The same can be done for `containerSize` and `innerSizes`.